eclipse/plugin/src/nl/tudelft/watchdog/eclipse/logic/ui/listeners/staticanalysis/MarkerHolder.java

```diff
@@ -0,0 +1,51 @@
+package nl.tudelft.watchdog.eclipse.logic.ui.listeners.staticanalysis;
+
+import org.eclipse.core.resources.IMarker;
+import org.eclipse.core.runtime.CoreException;
+import org.joda.time.DateTime;
+
+import nl.tudelft.watchdog.core.logic.ui.listeners.staticanalysis.StaticAnalysisMessageClassifier;
+
+/**
+ * Because an {@link IMarker} throws exceptions on {@link IMarker#getAttribute(String)} if it is removed,
+ * we have to extract all useful information at the first moment we encounter it, instead of invoking
+ * these methods once the marker is removed. At this moment, we are only interested in its message,
+ * as all other attributes are non-unique (yes even the ID is not unique). The only way to keep track
+ * of messages in a somewhat unique manner is its message, which does include private information.
+ * Make sure that when you use this message, the data is anonymized.
```

Comment:
A bit vague ...

Reply:
I am not sure what you find vague on this statement. I have reworded the sentence to my best knowledge of what you might misunderstand, PTAL.